api/v1beta2/types.go

```diff
@@ -356,12 +356,68 @@ type IBMCloudResourceReference struct {
 	Name *string `json:"name,omitempty"`
 }
 
+// IBMCloudCatalogOffering represents an IBM Cloud Catalog Offering resource.
+// +kubebuilder:validation:XValidation:rule="(has(self.offeringCRN) && !has(self.versionCRN)) || (!has(self.offeringCRN) && has(self.versionCRN))",message="either offeringCRN or version CRN must be provided, not both"
+type IBMCloudCatalogOffering struct {
+	// OfferingCRN defines the IBM Cloud Catalog Offering CRN. Using the OfferingCRN expects that the latest version of the Offering will be used.
+	// If a specific version should be used instead, rely on VersionCRN.
+	// +optional
+	OfferingCRN *string `json:"offeringCRN,omitempty"`
+
+	// PlanCRN defines the IBM Cloud Catalog Offering Plan CRN to use for the Offering.
+	// +optional
+	PlanCRN *string `json:"planCRN,omitempty"`
+
+	// VersionCRN defines the IBM Cloud Catalog Offering Version CRN. A specific version of the Catalog Offering will be used, as defined by this CRN.
+	// +optional
+	VersionCRN *string `json:"versionCRN,omitempty"`
+}
+
 // NetworkInterface holds the network interface information like subnet id.
 type NetworkInterface struct {
+	// SecurityGroups defines a set of IBM Cloud VPC Security Groups to attach to the network interface.
+	// +optional
+	// +k8s:conversion-gen=false
+	SecurityGroups []VPCResource `json:"securityGroups,omitempty"`
```

Comment:
Adding his field throwing error in the apidiff CI job - https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-ibmcloud/1978/pull-cluster-api-provider-ibmcloud-apidiff/1842074223984513024

Reply:
Per [comment above](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1978#discussion_r1787841017), I retried the manual conversion, so hopefully that resolves this failure now.